src/datasets/hf_dataset.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+from datasets import get_dataset_config_names, load_dataset
+from torch.utils.data import IterableDataset
+from tqdm import tqdm
+
+from src.utils.common import clean_text
+from src.utils.slicer import VTRSlicer
+
+
+class DatasetNLLB(IterableDataset):
+    def __init__(self, char2array: dict, window_size: int = 30, stride: int = 5, max_seq_len: int = 512):
+        self.datasets = dict()
+        self.slicer = VTRSlicer(char2array=char2array, window_size=window_size, stride=stride)
+        self.max_seq_len = max_seq_len
+        self.pairs = get_dataset_config_names("allenai/nllb")
+        self.lang2label: dict[str, int] = {}
+        self.label2lang: dict[int, str] = {}
+        label = 0
+        for pair in tqdm(self.pairs):
+            for lang in pair.split("-"):
+                if lang not in self.lang2label:
+                    self.lang2label[lang] = label
+                    self.label2lang[label] = lang
+                    label += 1
+            dataset = load_dataset("allenai/nllb", pair, split="train", streaming=True)
+            self.datasets[pair] = iter(dataset)
+
+        np.random.seed(0)
+
+    def __iter__(self):
+        while True:
+            random_pair = np.random.choice(self.pairs)
```

Comment:
And why do we want to iterate over the pairs in random order here? I would just iterate over the pairs in some predictable order.